src/java.base/share/classes/java/lang/foreign/MemorySegment.java

```diff
@@ -55,65 +56,81 @@
+ * <p>
+ * Native segments can be obtained by calling one of the {@link MemorySegment#allocateNative(long, long)}
+ * factory methods, which return a memory segment backed by a newly allocated off-heap region with the given size
+ * and alignment constraint. Alternatively, native segments can be obtained by
+ * {@link FileChannel#map(MapMode, long, long, MemorySession) mapping} a file into a new off-heap region
+ * (in some systems, this operation is sometimes referred to as {@code mmap}).
+ * Segments obtained in this way are called <em>mapped</em> segments, and their contents can be {@linkplain #force() persisted} and
+ * {@linkplain #load() loaded} to and from the underlying memory-mapped file.
+ * <p>
+ * Both kinds of segments are read and written using the same methods, known as <a href="#segment-deref">access operations</a>.
+ * An access operation on a memory segment always and only provides access to the region for which the segment was obtained.
  *
- * <h2 id="lifecyle-confinement">Lifecycle and confinement</h2>
+ * <h2 id="segment-characteristics">Characteristics of memory segments</h2>
  *
- * Memory segments are associated with a {@linkplain MemorySegment#session() memory session}. As for all resources associated
- * with a memory session, a segment cannot be accessed after its underlying session has been closed. For instance,
- * the following code will result in an exception:
- * {@snippet lang=java :
- * MemorySegment segment = null;
- * try (MemorySession session = MemorySession.openConfined()) {
- *     segment = MemorySegment.allocateNative(8, session);
- * }
- * segment.get(ValueLayout.JAVA_LONG, 0); // already closed!
- * }
- * Additionally, access to a memory segment is subject to the thread-confinement checks enforced by the owning memory
- * session; that is, if the segment is associated with a shared session, it can be accessed by multiple threads;
- * if it is associated with a confined session, it can only be accessed by the thread which owns the memory session.
+ * Every memory segment has an {@linkplain #address() address}, expressed as a {@code long} value.
+ * The nature of a segment's address depends on the kind of the segment:
+ * <ul>
+ * <li>The address of a heap segment is not a physical address, but rather an offset within the region of memory
+ * which backs the segment. The region is inside the Java heap, so garbage collection might cause the region to be
+ * relocated in physical memory over time, but this is not exposed to clients of the {@code MemorySegment} API who
+ * see a stable <em>virtualized</em> address for a heap segment backed by the region.
+ * A heap segment obtained from one of the {@link #ofArray(int[])} factory methods has an address of zero.</li>
+ * <li>The address of a native segment (including mapped segments) denotes the physical address of the region of
+ * memory which backs the segment.</li>
```

Comment:
I get what is meant by it, but I don't think "physical" is a great term wrt intuitive meaning. It would maybe be nice to define the term somewhere up front if we want to stick with "physical".

Reply:
I think for regular Java developers, there is probably a much less clear distinction between heap/off-heap and "physical" memory in the first place.

i.e. if heap memory is the part of memory that is used to store Java objects, and off-heap is the part of memory that _isn't_ used to store Java objects, then I guess "physical memory" is just all of memory combined? Maybe just saying "memory" is enough? In the above, that could make:

    The {@linkplain #address() address} of a native segment (including mapped segments) denotes the starting address of the region of memory which backs the segment.

Reply:
If none of these options seem better to you though, I don't mind sticking with "physical address/memory".

Reply:
I think `physical memory` is the problem, more so than `physical address`. We can explain `physical address` as something different from `MemorySegment::address`, but I don't think we need a qualifier for memory.

Reply:
I looked at this again, and decided to leave as is. `MappedByteBuffer` is also quite full of references to terms like `physical memory` (which we inherited in methods like `MemorySegment::isLoaded`.